components/esp_wifi_remote/eppp/wifi_remote_rpc_impl.hpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+#include <cstring>
+#include <cerrno>
+
+namespace eppp_rpc {
+
+const int rpc_port = 3333;
+
+enum class api_id : uint32_t {
+    ERROR,
+    UNDEF,
+    INIT,
+    SET_MODE,
+    SET_CONFIG,
+    START,
+    STOP,
+    CONNECT,
+    GET_MAC,
+    WIFI_EVENT,
+    IP_EVENT,
+};
+
+enum class role {
+    SERVER,
+    CLIENT,
+};
+
+struct RpcHeader {
+    api_id id;
+    uint32_t size;
+} __attribute((__packed__));
+
+template<typename T>
+struct RpcData {
+    RpcHeader head;
+    T value_{};
+    explicit RpcData(api_id id) : head{id, sizeof(T)} {}
+
+    uint8_t *value()
```

Comment:
This could be non-member `operator*`

Reply:
Or hidden friend

Reply:
i'll stick to the current member function name, as the operator* would be confusing and might indicate that we're dereferencing the instance pointer ( rpc->value() )